### PR TITLE
feat(spanner): add samples for creating instances with processing units

### LINKED ISSUE #793: spanner: example to create low cost instance using processing unit

### PR DESCRIPTION
**Pull request by jiren**
## Description

Fixes #793 793

## Checklist
- [x] **Tests** pass
- [x] **Lint** pass: `bundle exec rubocop`
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/spec/spanner_samples_spec.rb [hunk 1]
```diff
@@ -148,6 +148,22 @@ def capture
...
+    capture do
+      create_instance project_id:  @project_id,
+                      instance_id: @created_instance_id
+    end
+
+    expect(captured_output).to include(
+      "Waiting for create instance operation to complete"
+    )
+    expect(captured_output).to include(
+      "Instance #{@created_instance_id} has 1000 processing units."
```
Comment (hengfengli) [inline spanner/spec/spanner_samples_spec.rb:163]: 1000 -> 500?

File: spanner/spanner_samples.rb [hunk 2]
```diff
@@ -40,6 +40,38 @@ def create_instance project_id:, instance_id:
...
 
+def create_instance_with_processing_units project_id:, instance_id:
+  # [START spanner_create_instance_with_processing_units]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
```
Comment (hengfengli) [inline spanner/spanner_samples.rb:51]: delete?

File: spanner/spec/spanner_samples_spec.rb [hunk 3]
```diff
@@ -148,6 +148,22 @@ def capture
...
+  example "create_instance_with_processing_units" do
+    @created_instance_id = "rb-test-#{seed}"
+
+    capture do
+      create_instance project_id:  @project_id,
+                      instance_id: @created_instance_id
+    end
+
+    expect(captured_output).to include(
+      "Waiting for create instance operation to complete"
```
Comment (hengfengli) [inline spanner/spec/spanner_samples_spec.rb:160]: create -> creating?

File: spanner/spanner_samples.rb [hunk 4]
```diff
@@ -40,6 +40,38 @@ def create_instance project_id:, instance_id:
...
+
+  job.wait_until_done!
+
+  if job.error?
+    puts job.error
+  else
+    puts "Created instance #{instance_id}"
+  end
+
+  instance = spanner.instance instance_id
```
Comment (hengfengli) [inline spanner/spanner_samples.rb:69]: @jiren According to the spec, we need to call `getInstance` and show users how to use `fieldMask`. Can you please update the sample accordingly?

Comment (jiren) [reply spanner/spanner_samples.rb:69]: There is no specific method to fetch instance with fieldmask in Ruby lib. 

Only following method is available.
```
instance = spanner.instance instance_id
```

Comment (hengfengli) [reply spanner/spanner_samples.rb:69]: Okay. That sounds good to me.

**Review by hengfengli — APPROVED** [review]

**Comment by hengfengli** [comment]
@dazuma Can you please help us to merge this PR? Thanks.

File: spanner/spanner_samples.rb (hunk 2 above)
Comment (dazuma) [reply spanner/spanner_samples.rb:51]: This `instance` variable is never used. I think this line should be deleted.

**Review by dazuma — CHANGES_REQUESTED** [review]

**Comment by hengfengli** [comment]
@dazuma Please take a look again. Thanks!

**Review by dazuma — APPROVED** [review]

**Pull request merged**